torchvision/transforms/v2/functional/_meta.py

```diff
@@ -296,7 +296,10 @@ def _num_value_bits(dtype: torch.dtype) -> int:
         raise TypeError(f"Number of value bits is only defined for integer dtypes, but got {dtype}.")
 
 
-def convert_dtype_image_tensor(image: torch.Tensor, dtype: torch.dtype = torch.float) -> torch.Tensor:
+def to_dtype_image_tensor(image: torch.Tensor, dtype: torch.dtype = torch.float, scale: bool = False) -> torch.Tensor:
+    if not scale:
```

Comment:
Should this come after the dtype check? Functionally, this is a no-op here in case `image.dtype == dtype`, but it is still an unnecessary call.

```python
if image.dtype == dtype:
    return image
elif not scale:
    return image.to(dtype)
```

makes the behavior a little more clear and should be minimally more performant.

Reply:
I'll do it but TBH  I don't really follow the reasoning behind the change

Reply:
The `elif` part actually adds confusion to me because it suggests these 2 blocks are related when in reality they're really not

Reply:
Re-reading the original implementation: I was wrong. Since we already `return`'ed in both branches, this should have no effect on the performance. Thus, feel free to revert to what you had.

Still, I would put the dtype check at the top, because that is the "more important" one. Again, just style / personal preference. Ignore if you feel different.



> The `elif` part actually adds confusion to me because it suggests these 2 blocks are related when in reality they're really not

Fair enough. Happy with a second `if` as well.